### PR TITLE
Eta-expand hints

### PR DESCRIPTION
**Pull request by josephcsible**
Eta-expand instead of using composition, so that we can match them both ways.

### CODE REVIEW

File: data/hlint.yaml [hunk 1]
```diff
@@ -278,15 +278,15 @@
...
     - warn: {lhs: (f $), rhs: f, name: Redundant $}
     - warn: {lhs: (Data.Function.& f), rhs: f, name: Redundant Data.Function.&}
     - hint: {lhs: \x -> y, rhs: const y, side: isAtom y && not (isWildcard y)}
         # If any isWildcard recursively then x may be used but not mentioned explicitly
     - warn: {lhs: flip f x y, rhs: f y x, side: isApp original}
     - warn: {lhs: id x, rhs: x}
-    - warn: {lhs: id . x, rhs: x, name: Redundant id}
-    - warn: {lhs: x . id, rhs: x, name: Redundant id}
+    - warn: {lhs: id (f x), rhs: f x, name: Redundant id}
+    - warn: {lhs: f (id x), rhs: f x, name: Redundant id}
```
Comment (ndmitchell) [inline data/hlint.yaml]: There is already a rule `id x ==> x`. I think this ends up duplicating that?

File: data/hlint.yaml [hunk 2]
```diff
@@ -278,15 +278,15 @@
...
+    - warn: {lhs: ($) (f x), rhs: f x, name: Redundant $}
     - warn: {lhs: (f $), rhs: f, name: Redundant $}
     - warn: {lhs: (Data.Function.& f), rhs: f, name: Redundant Data.Function.&}
     - hint: {lhs: \x -> y, rhs: const y, side: isAtom y && not (isWildcard y)}
         # If any isWildcard recursively then x may be used but not mentioned explicitly
     - warn: {lhs: flip f x y, rhs: f y x, side: isApp original}
     - warn: {lhs: id x, rhs: x}
-    - warn: {lhs: id . x, rhs: x, name: Redundant id}
-    - warn: {lhs: x . id, rhs: x, name: Redundant id}
+    - warn: {lhs: id (f x), rhs: f x, name: Redundant id}
```
Comment (ndmitchell) [inline data/hlint.yaml]: Dupe of `id x ==> x` ?

**Comment by josephcsible** [comment]
Fixed, and merge conflicts resolved.

**Comment by ndmitchell** [comment]
Thanks!

**Pull request merged**